faunadb-java-dsl/src/main/java/com/faunadb/client/query/Language.java

```diff
@@ -1099,6 +1102,11 @@ public static Expr Map(Expr collection, Expr lambda) {
     return Fn.apply("map", lambda, "collection", collection);
   }
 
+  public static Expr Map(Expr collection, UnaryOperator<Expr> lambda) {
+    Expr sym = Var(genSym("map"));
```

Comment:
what if you get the lambda name by reflection? `lambda.getClass().getMethod("apply", Expr.class).getParameters()[0].getName()`

Reply:
ClassValue. Neat!
I didn't know that class existed.

Reply:
Heh. I jumped through those very hoops, and thought it seemed like overhead for another day. @erick and I had a bit of back-and-forth on the ticket.

This solution avoids reflection, but still provides some hint as to where an error might be coming from to the user.

I seem to be in the minority about reflection, though. :)

Reply:
Yeah. I feel like it's nice to preserve the variable name but, I don't think it's really worth the additional runtime cost.

Reply:
Honestly, I'm not too hot on parameter names either. I guess I lived with Java for long enough without having reflective access to names of parameters. I still feel like they aren't really useful as public symbols; they're basically variable names local to the method implementation and as such belong in the realm of debug information. It can even be dangerous if people come to rely on them having particular values.